plugins/plugin.go

```diff
@@ -0,0 +1,79 @@
+package plugins
+
+import (
+	"sync"
+)
+
+const (
+	// HandShakePath is the path for handshaking
+	HandShakePath = "/Plugin.Activate"
+)
+
+// TLSConfig contains information of tls which users can specify
+type TLSConfig struct {
```

Comment:
I think we have already defined a TLSConfig in the code base package `client`. I am wondering if we could avoid the duplication.

Reply:
plugins.TLSConfig is different from client.TLSConfig